app/controllers/admin/users_controller.rb

```diff
@@ -28,6 +28,7 @@ def destroy
     #
     def scoped_resource
       return super if params[:search].present?
+      return resource_class if action_name == "destroy"
```

Comment:
```suggestion
      # Unconfirmed users can be deleted, but are otherwise not included when administering users
      # We include them during destroy so that they can be found to be destroyed.
      return resource_class if action_name == "destroy"
```